src/GraphQL/transformers/className.js

```diff
@@ -0,0 +1,39 @@
+import logger from '../../logger';
+
+const parseMap = {
+  Role: '_Role',
+  User: '_User',
+  Session: '_Session',
+  // Add native type to avoid name collision
+  Files: 'Files',
+  String: 'String',
+  Boolean: 'Boolean',
+  Number: 'Number',
+  Array: 'Array',
+  Object: 'Object',
+  Date: 'Date',
+  GeoPoint: 'GeoPoint',
+  Polygon: 'Polygon',
+  Bytes: 'Bytes',
+};
+const reverseParseMap = {};
+// Dynamically fill reverseParseMap based on parseMap
+Object.keys(parseMap).forEach(key => (reverseParseMap[parseMap[key]] = key));
+
+const transformClassNameToParse = className => {
+  if (reverseParseMap[className])
+    logger.error(
+      `Class collision detected, please change the name of your class ${className}`
+    );
+  return parseMap[className] ? parseMap[className] : className;
+};
+
+const transformClassNameToGraphQL = className => {
```

Comment:
I also don't like so much the idea of transform to parse and then transform to GraphQL. I think it can bring some problems when we start growing this. I think a better approach would be just transform to GraphQL and save the two versions in parseGraphQLSchema.parseClasses.

Reply:
Transforms functions acts here as a translate layer (in one place), and ensure that no collision can happen after a translation (Parse2GQL/GQL2Parse).
I'm not sure to understand the duplication in `parseGraphQLSchema.parseClasses` ?

Can you provide a small example ?

Reply:
An example would be the developer creating its own custom class called `Installation` instead of using the default `_Installation` one. Instead of transforming and reversing, I'd suggest to remove any `_` in the beginning and also upper case the first letter (in the case that the developer has created a class called `car`). We don't need to reverse the name. You just need to always use the original parse class name stored in `parseGraphQLSchema.parseClasses` when passing to Parse.